dask/array/slicing.py

```diff
@@ -302,6 +302,13 @@ def slice_slices_and_integers(out_name, in_name, blockdims, index):
                 f"Arrays chunk sizes are unknown: {shape}{unknown_chunk_message}"
             )
 
+    def normalize_np_scalars(ind):
```

Comment:
I'm not sure defining a function within another function is a good idea here. 

Can we take this function out, and put it together with the other `normalize_` functions?  Going even further do we need a separate function or can we extend `normalize_index` or `normalize_slice`

Reply:
I don't think having a nested function is inherently a bad thing. That said, in this particular case I agree it's probably not needed. @fAndreuzzi I think something like

```diff
diff --git a/dask/array/slicing.py b/dask/array/slicing.py
index 05ab32735..b9a6b59fb 100644
--- a/dask/array/slicing.py
+++ b/dask/array/slicing.py
@@ -302,12 +302,8 @@ def slice_slices_and_integers(out_name, in_name, blockdims, index):
                 f"Arrays chunk sizes are unknown: {shape}{unknown_chunk_message}"
             )

-    def normalize_np_scalars(ind):
-        if isinstance(ind, np.ndarray) and ind.ndim == 0:
-            ind = ind.flatten()[0]
-        return ind
-
-    index = tuple(map(normalize_np_scalars, index))
+    # Extract 0-dimension ndarrays
+    index = [i.item() if is_arraylike(i) and i.ndim == 0 else i for i in index]

     assert all(isinstance(ind, (slice, Integral)) for ind in index)
     assert len(index) == len(blockdims)
```

should have the same effect

Reply:
@ncclementi @jrbourbeau Thanks for your input! Indeed it was a bad design choice, fixed in 2d79dd4ee39992c49b2cc42ac36fa747ce5d82e2